gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/push/PushMessageValidator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.push;
+
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.validation.validator.constraints.ConstraintValidator;
+import io.micronaut.validation.validator.constraints.ConstraintValidatorContext;
+import jakarta.inject.Singleton;
+
+/**
+ * A validator for incoming PubSub push messages.
+ *
+ * @author Jeremy Grelle
+ * @since 5.4.0
+ */
+@Singleton
```

Comment:
why do you need the validator to be `@Singleton` you are not using any collaborator to perform the validation?

Reply:
Another documentation issue - my interpretation of the docs here were that it is required to be a bean from the context: https://micronaut-projects.github.io/micronaut-validation/latest/guide/index.html#additionalConstraints

I have run the tests without it and verified it is unnecessary, so removing.

Reply:
[cbcd2c8](https://github.com/micronaut-projects/micronaut-gcp/pull/1012/commits/cbcd2c88e75eaeb474af6d9031869e8ceef8a4c6)